.jenkins/pytorch/test.sh

```diff
@@ -259,34 +259,50 @@ test_inductor() {
   PYTORCH_TEST_WITH_INDUCTOR=0 python test/run_test.py --include inductor/test_torchinductor --include inductor/test_torchinductor_opinfo --verbose
 }
 
-test_inductor_benchmark() {
+test_single_dynamo_benchmark() {
+  # Usage: test_single_dynamo_benchmark inductor_inference huggingface 0 --args-for-script
+
   # Use test-reports directory under test folder will allow the CI to automatically pick up
   # the test reports and upload them to S3. Need to use full path here otherwise the script
   # will bark about file not found later on
   TEST_REPORTS_DIR=$(pwd)/test/test-reports
-  PARTITION_FLAGS=""
-  if [[ -n "$NUM_TEST_SHARDS" && -n "$2" ]]; then
-    PARTITION_FLAGS="--total-partitions 2 --partition-id $2"
-  fi
   mkdir -p "$TEST_REPORTS_DIR"
+
+  local name="$1"
+  shift
+  local suite="$1"
+  shift
+  # shard id is mandatory, even if it is not passed
+  local shard_id="$1"
+  shift
+
+  local partition_flags=()
+  if [[ -n "$NUM_TEST_SHARDS" && -n "$shard_id" ]]; then
+    partition_flags=( --total-partitions 2 --partition-id "$shard_id" )
+  fi
+
+  # Feel free to remove --device cuda if you ever decide to need to
+  # test CPU as well in CI
```

Comment:
cc @jgong5 @EikanWang , FYI, since you are working on adding CPU benchmark test.